### PR TITLE
Allow existing user to self-serve new invite

### PR DESCRIPTION
**Pull request by alex-yau-ttd**
- New unauthenticated route and endpoint
- Add link to the invite email that takes the user to the new route, prefilled with their email address
- Some refactoring in `configureApi.ts` to improve readability

**Demo**

https://github.com/IABTechLab/uid2-self-serve-portal/assets/137864392/ec0832fe-03bc-48db-b24b-5da1d6f1dad4

### CODE REVIEW

File: src/api/configureApi.ts [hunk 1]
```diff
@@ -46,13 +46,24 @@ type BypassPath = {
   method: string;
 };
 
-const BYPASS_PATHS = [
-  `/favicon.ico`,
-  `${BASE_REQUEST_PATH}/`,
-  `${BASE_REQUEST_PATH}/metrics`,
-  `${BASE_REQUEST_PATH}/health`,
-  `${BASE_REQUEST_PATH}/keycloak-config`,
-].map((path) => ({ url: path, method: 'GET' }));
```
Comment (alex-yau-ttd) [inline src/api/configureApi.ts:55]: I thought this was harder to read and makes it tricky when you want to add a `POST` method

File: src/api/controllers/userController.ts [hunk 2]
```diff
@@ -77,6 +82,23 @@ export class UserController {
...
+  @httpPost('/selfResendInvitation')
+  public async selfResendInvitation(
+    @request() req: SelfResendInviteRequest,
+    @response() res: Response
+  ): Promise<void> {
+    const { email } = SelfResendInvitationParser.parse(req.body);
+    const logger = this.loggerService.getLogger(req);
+    const kcAdminClient = await getKcAdminClient();
+    const user = await queryUsersByEmail(kcAdminClient, email);
+    if (user.length !== 1) {
```
Comment (alex-yau-ttd) [inline src/api/controllers/userController.ts:94]: This is the "unhappy" case, be we just want to return 200 here.

File: keycloak/themes/uid2-theme/email/messages/messages_en.properties [hunk 3]
```diff
@@ -4,5 +4,5 @@ emailVerificationBody=Please verify your email address\n\nClick the link below t
...
 passwordResetBody=Hi {0}, We received a request to reset your password for your account. Please use the link below to reset your password.{1}
-inviteBodyHtml=<p> Hi {0}, </p> <p> You’ve been added to the UID2 Portal. </p> <p> Click the button to accept your invitation and create a password for your account. </p> <p>Note: This invitation expires in {2}.</p> <div class="footer"> <a href="{1}" class="button" >Accept Invitation</a > </div>
+inviteBodyHtml=<p> Hi {0}, </p> <p> You’ve been added to the UID2 Portal. </p> <p> Click the button to accept your invitation and create a password for your account. </p> <p> If you didn’t request access, you can ignore this message. </p> <p>Note: This invitation expires in {2}.</p> <div class="footer"> <a href="{1}" class="button" >Accept Invitation</a > <p> <br /> If the invitation has expired, <a href="http://localhost:3000/selfReinvite?email={3}" class="link">request a new invitation</a>. </p> </div>
```
Comment (lionell-pack-ttd) [inline keycloak/themes/uid2-theme/email/messages/messages_en.properties:7]: You still have `localhost:3000` in the template.

Comment (alex-yau-ttd) [reply keycloak/themes/uid2-theme/email/messages/messages_en.properties]: Fixed!

**Pull request merged**